model/pipeline/src/main/resources/org/apache/beam/model/pipeline/pipeline.json

```diff
@@ -0,0 +1,320 @@
+        // and the tag should contain sufficient information to identify the target PCollection.
+        "inputs": { "type": "array", "items": { "type": "string" } },
+        "outputs": { "type": "array", "items": { "type": "string" } },
+
+        // If the node is contained within a composite, a pointer to the parent.
+        "parent": { "type": [ "null", "string" ] },
+
+        // If the node is a runner-specific replacement for a user's runner-agnostic node,
+        // it contains a pointer back to it so the user's graph can be reconstructed.
+        "replacementFor": { "type": [ "null", "string" ] },
+
+        // If the node is a defunct user node that has been replaced, it contains a bit
+        // to distinguish it from an active node in the graph.
+        "replaced": { "type": "boolean" },
+
+        // Static display data
+        "display_data": { "type": "object", "additionalProperties": { "type": "string" } },
+
+        // The payload must correspond to the kind string and must be null for transforms that
+        // are not predefined as part of the Beam model.
+        "payload": {
+          "type": [ "null", "object" ],
+          "oneOf": [
+            { "$ref": "#/definitions/ParDoPayload" },
+            { "$ref": "#/definitions/ReadPayload" },
+            { "$ref": "#/definitions/WindowIntoPayload" },
+            { "$ref": "#/definitions/CombinePayload" }
+          ]
+        }
+      }
+    },
+
+    // A value node represents a PValue. There are only two primitive types:
+    // A PCollection (many values per window) or a PCollectionView (one value per window).
+    "Value": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        // A unique program-readable name for the value, such as "foo/baz/bar.out
+        "label": { "type": "string" },
+
+        // The node ID for the coder of this PCollection
+        "coder": { "type": "string" },
+
+        // Windowing strategy node id
+        "windowingStrategy": { "type": "string" },
+
+        // This is the static display data recently proposed
+        "displayData": { "$ref": "#/definitions/DisplayData" }
+      }
+    },
+
+    "DisplayData": {
+      "type": "object",
+      "additionalProperties": { "type": "string" }
+    },
+
+    // An coder represents a particular binary format.
+    "Coder": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+
+        // A cross-language, stable, unique identifier for the (possibly parametric) encoding.
+        "urn": { "type": "string" },
+
+        // If this coder is parametric, such as ListCoder(VarIntCoder), this is a list
+        // of the components. In order for encodings to be identical, the encoding_id
+        // and all components must be identical.
+        "componentEncodings": { "type": "array", "items": { "type": "string" } },
```

Comment:
Yes. The type was right - array of string ids - but the comment is not clear about that.